ledger-api/grpc-definitions/com/daml/ledger/api/v2/state_service.proto

```diff
@@ -0,0 +1,152 @@
+
+option java_outer_classname = "StateServiceOuterClass";
+option java_package = "com.daml.ledger.api.v2";
+option csharp_namespace = "Com.Daml.Ledger.Api.V2";
+
+// TODO add documentation of bootstrapping procedure
+// Allows clients to get state from the ledger.
+service StateService {
+
+  // Returns a stream of the snapshot of the active contracts at a ledger offset.
+  // If there are no active contracts, the stream returns a single response message with the offset at which the snapshot has been taken.
+  // Clients SHOULD use the offset in the last GetActiveContractsResponse message to continue streaming transactions with the update service.
+  // Clients SHOULD NOT assume that the set of active contracts they receive reflects the state at the ledger end.
+  rpc GetActiveContracts (GetActiveContractsRequest) returns (stream GetActiveContractsResponse);
+
+  // Get the list of connected domains at the time of the query.
+  rpc GetConnectedDomains (GetConnectedDomainsRequest) returns (GetConnectedDomainsResponse);
+
+  // Get the current ledger end.
+  // Subscriptions started with the returned offset will serve events after this RPC was called.
+  rpc GetLedgerEnd (GetLedgerEndRequest) returns (GetLedgerEndResponse);
+}
+
+message GetActiveContractsRequest {
+
+  // Templates to include in the served snapshot, per party.
+  // Required
+  TransactionFilter filter = 1;
+
+  // If enabled, values served over the API will contain more information than strictly necessary to interpret the data.
+  // In particular, setting the verbose flag to true triggers the ledger to include labels for record fields.
+  // Optional
+  bool verbose = 2;
+
+  // The offset at which the snapshot of the active contracts will be computed.
+  // Must be no greater than the current ledger end offset.
+  // Must be greater than or equal to the last pruning offset.
+  // If not set the current ledger end offset will be used.
+  // Optional
+  string active_at_offset = 3;
+}
+
+message GetActiveContractsResponse {
+
+  // Included only in the last message.
+  // The client should start consuming the transactions endpoint with this offset.
+  // The format of this field is described in ``ledger_offset.proto``.
+  string offset = 1;
+
+  // For each contract there could be multiple ContractStateComponent-s in the entire snapshot. These together define
+  // the state of the contract in the snapshot.
+  ContractStateComponent contract_state_components = 2;
+}
+
+message ContractStateComponent {
+  oneof contract_state_component {
+    // The contract is active in the given domain.
+    // There will be one such message for each domain the contract is active on.
+    ActiveContract active_contract = 1;
+
+    // This message is only available, if the corresponding assigned event cannot be found before or at the
+    // active_at_offset, and at least one of the querying parties could submit the corresponding assign command.
```

Comment:
I like it a lot, thanks!

Reply:
Oh wait: you probably accidentally forgot the last part, right? So it would look like this:
"Included iff the unassigned event was before or at the active_at_offset but there was no corresponding assigned event before or at the active_at_offset, and at least one of the querying parties could submit the corresponding assign command."

Reply:
yeah forgot the last part.

Reply:
changed accordingly